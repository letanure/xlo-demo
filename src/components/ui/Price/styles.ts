import styled, { css } from 'styled-components'

import { PriceProps } from '.'

export const Wrapper = styled.div<PriceProps>`
  ${() => css``}
`
