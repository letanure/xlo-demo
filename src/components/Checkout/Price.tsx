import * as S from './styles'

export interface Props {
  value: number
  lineThrough?: boolean
  hasAsterisk?: boolean
}

const Price = ({ value, lineThrough, hasAsterisk }: Props) => {
  const formatNumber = (value: number) =>
    new Intl.NumberFormat('pl-PL', {
      style: 'currency',
      currency: 'PLN'
    })
      .format(value)
      .replace(',00', '')
  return (
    <S.Price lineThrough={lineThrough}>
      <S.PriceValue>{formatNumber(value)}</S.PriceValue>
      {hasAsterisk && '*'}
    </S.Price>
  )
}

export default Price
