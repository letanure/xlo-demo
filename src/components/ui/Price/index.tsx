import * as S from './styles'

// ISO 4217 Currency Code
type CurrencyCode = 'PLN' | 'RON'

export type PriceProps = {
  amount: number
  currency: CurrencyCode
}

type LanguageCodesObject = {
  [key in CurrencyCode]: string
}

const languageCodes: LanguageCodesObject = {
  PLN: 'pl-PL',
  RON: 'ro-RO'
}

const Price = ({ amount, currency }: PriceProps) => {
  const formatedPrice = new Intl.NumberFormat(
    languageCodes[currency as CurrencyCode],
    {
      style: 'currency',
      currency: currency
    }
  ).format((amount as number) / 100)

  return <S.Wrapper>{formatedPrice}</S.Wrapper>
}

export default Price
