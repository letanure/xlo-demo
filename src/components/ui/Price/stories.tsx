import { ComponentStory, ComponentMeta } from '@storybook/react'

import Price from '.'

export default {
  title: 'UI/Price',
  component: Price,
  argTypes: {
    amount: { control: 'number' },
    currency: { control: 'select' }
  }
} as ComponentMeta<typeof Price>

const Template: ComponentStory<typeof Price> = (args) => <Price {...args} />

export const Basic = Template.bind({})
Basic.args = {
  amount: 1000,
  currency: 'PLN'
}
