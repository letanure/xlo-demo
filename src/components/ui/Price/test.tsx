import { render, screen } from '@testing-library/react'

import Price from '.'

describe('<Price />', () => {
  it.skip('should render something', () => {
    const { container } = render(<Price />)

    expect(screen.getByRole('heading', { name: /Price/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
