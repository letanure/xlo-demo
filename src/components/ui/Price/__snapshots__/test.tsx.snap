// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> should render something 1`] = `
.c0 {
  content: 'default';
}

<div
  class="c0"
>
  <h1>
    Price with 
    default
  </h1>
</div>
`
